debezium-server/debezium-server-redis/src/test/java/io/debezium/server/redis/RedisStreamIT.java

```diff
@@ -61,15 +74,111 @@ void connectorCompleted(@Observes ConnectorCompletedEvent event) throws Exceptio
+                "INSERT INTO inventory.redis_test VALUES (5)");
+        connection.close();
+
+        Testing.print("Sleeping for 5 seconds to simulate no connection errors");
+        Thread.sleep(5000);
+        Testing.print("Unpausing container");
+        RedisTestResourceLifecycleManager.unpause();
+
+        final List<StreamEntry> entries = new ArrayList<>();
+        Awaitility.await().atMost(Duration.ofSeconds(RedisTestConfigSource.waitForSeconds())).until(() -> {
+            final List<StreamEntry> response = jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT);
+            entries.clear();
+            entries.addAll(response);
+            return entries.size() >= MESSAGE_COUNT;
+        });
+        Testing.print("Entries in " + STREAM_NAME + ":" + entries.size());
+        assertTrue("Redis Connection Test Failed", entries.size() == MESSAGE_COUNT);
+    }
+
+    @Test
+    @FixFor("DBZ-4510")
+    public void testRedisOOMRetry() throws Exception {
+        /*
+         * Test retry mechanism when encountering Redis Out of Memory:
+         * 1. Simulate a Redis OOM by setting its max memory to 1M.
+         * 2. Create a new table named redis_test2 in PostgreSQL and insert 1000 records to it.
+         * 3. As result, after inserting ~22 records, Redis runs OOM.
+         * 4. Sleep for additional 5 seconds to ensure the Sink is retrying.
+         * 5. Revert max memory setting so Redis is no longer in OOM and make sure all 100 records have been streamed successfully.
+         */
+
+        final int MESSAGE_COUNT = 100;
+        final String STREAM_NAME = "testc.inventory.redis_test2";
+        Testing.print("Setting Redis' maxmemory to 1M");
+        jedis.configSet("maxmemory", "1M");
+        final PostgresConnection connection = getPostgresConnection();
+        Testing.print("Creating new table redis_test2 and inserting 100 records to it");
+        connection.execute(
+                "CREATE TABLE inventory.redis_test2 (id varchar(100) PRIMARY KEY, first_name varchar(100), last_name varchar(100))",
+                "insert into inventory.redis_test2 (id,first_name,last_name) select left(i::text, 10), random()::text, random()::text from generate_series(1,100) s(i)");
+        connection.close();
+
+        Testing.print("Sleeping for 5 seconds to simulate no connection errors");
+        Thread.sleep(5000);
+        Testing.print("Stream size in OOM:" + jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT).size());
+        Testing.print("Reverting Redis' maxmemory");
+        jedis.configSet("maxmemory", "0");
+
+        final List<StreamEntry> entries = new ArrayList<>();
+        Awaitility.await().atMost(Duration.ofSeconds(RedisTestConfigSource.waitForSeconds())).until(() -> {
+            final List<StreamEntry> response = jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT);
+            entries.clear();
+            entries.addAll(response);
+            return entries.size() >= MESSAGE_COUNT;
```

Comment:
Why is it >= and not =?

Reply:
You're right, I've changed it - I think that this is based on all the other tests of the other sinks, but it's definitely better to use == in this case.
BTW, I've changed the Redis' tests to use assertTrue instead of Assertions.assertThat since the later does not fail the test if the assertion is false (as in all the other tests in the assortment of sinks) - I think you should consider changing it there too as it's a bit weird that the tests do not fail in such cases.

Reply:
Ah yes, good catch. This usage of `assertThat` wasn't correct, it must be something like `assertThat(foo).eEqualsTrue()`. I .e. the actual assertion method was missing.